### ISSUE TITLE
Go to File doesn't find my JS files

### ISSUE THREAD

**Issue by borekb** [Microsoft/vscode#32638]
**UPDATE:** this might be related to JavaScript files and my `files.exclude` settings, see https://github.com/Microsoft/vscode/issues/32638#issuecomment-322768797.

I've got a folder open which has many sub-folders, many of them containing `gulpfile.js`. However, when I do Ctrl+P (Go to File), they are not listed.

In the screenshot, I've executed a search alongside Ctrl+P to demonstrate this:

![image](https://user-images.githubusercontent.com/101152/29364664-8d7abe5c-8294-11e7-9cb8-2e80386d136c.png)

This is the first time I see this issue, Ctrl+P has always been working reliably. I've tried restarting VSCode to no avail.

VSCode version: 1.15.0
OS: Windows 10

**Comment by borekb** [comment]
BTW I've tried for other files which come in larger quantities in our repo and all is fine there, it's just gulpfiles that are weird.

![image](https://user-images.githubusercontent.com/101152/29364970-9980fdf0-8295-11e7-9974-d1d41c4d2f0e.png)

![image](https://user-images.githubusercontent.com/101152/29364999-b13cc51e-8295-11e7-89be-c354406a2bc3.png)

![image](https://user-images.githubusercontent.com/101152/29365039-d4175676-8295-11e7-8493-a91a9da9526a.png)

**Comment by borekb** [comment]
It seems to be JS files, for example, Ctrl+P doesn't find my `webpack.js` file either.

This has been in my user settings for a very long time:

```json
{
    "files.exclude": {
        "**/.git": true,
        "**/.svn": true,
        "**/.hg": true,
        "**/.DS_Store": true,
        "**/*.js": {
            "when": "$(basename).ts"
        },
        "**/*?.js": {
            "when": "$(basename).tsx"
        },
        "**/*.js.map": {
            "when": "$(basename)"
        }
    },
    "search.exclude": {
        "**/node_modules": true,
        "**/bower_components": true,
        "**/dist": true,
        "**/user-portal/client/public": true
    }
}
```

Is it possible that the "when" resolution changed in 1.15?

**Comment by roblourens** [comment]
This looks like https://github.com/Microsoft/vscode/issues/31956, which is fixed in insiders, and will be fixed in the upcoming 1.15.1. Sorry for the trouble.

**Comment by borekb** [comment]
Cool, thanks.
